packages/sandbox/src/lambda_function_log_streamer.ts

```diff
@@ -54,33 +46,26 @@ export class LambdaFunctionLogStreamer {
       ? (JSON.parse(definedFunctionsPayload) as string[])
       : [];
 
-    // To use list-tags API we need to convert function name to function Arn since it only accepts ARN as input
-    const deployedFunctionNameToArnMap = await this.getFunctionArnFromNames(
-      sandboxBackendId,
-      deployedFunctionNames
-    );
-
-    if (!deployedFunctionNameToArnMap) {
-      this.printer.log(
-        `[Sandbox] Could not find any function in stack ${BackendIdentifierConversions.toStackName(
-          sandboxBackendId
-        )}. Streaming function logs will be turned off.`,
-        LogLevel.DEBUG
-      );
-      return;
-    }
-
-    for (const entry of deployedFunctionNameToArnMap) {
-      const listTagsResponse = await this.lambda.send(
-        new ListTagsCommand({
-          Resource: entry.arn,
+    for (const functionName of deployedFunctionNames) {
+      const getFunctionResponse = await this.lambda.send(
+        new GetFunctionCommand({
+          FunctionName: functionName,
```

Comment:
We already do have permissions to call `GetFunction` in `AmplifyBackendDeployFullAccess`

Reply:
The reason we went with `ListTags` earlier was for performance reasons, since it will be called after each sandbox deployment.

Reply:
Was `ListTags` significantly faster than `GetFunction` ?

Some considerations:
1. The number of requests to lambda CP doesn't change. We were calling `ListTags` per each function.
2. We eliminated extra CloudFormation calls to resolve ARNs.
3. The alternative would be to call `GetFunctionConfiguration` in addition to `ListTags`. But, making two requests is not great quota and latency wise. 
4. It seems that default quota for `GetFunction` is much more generous than other CP APIs. See picture below.

![image](https://github.com/user-attachments/assets/96f0be22-54f5-4187-a3c5-0529b688c4e3)

Reply:
Synced offline. We'll keep `GetFunction` given that alternatives are not better.